packages/select/src/components/multi-select/multiSelect.tsx

```diff
@@ -220,7 +230,22 @@ export class MultiSelect<T> extends AbstractPureComponent<MultiSelectProps<T>, M
                 {...popoverProps}
                 className={classNames(listProps.className, popoverProps.className)}
                 content={
-                    <div {...popoverContentProps} onKeyDown={handleKeyDown} onKeyUp={handleKeyUp}>
+                    <div
+                        // In the case where customTarget is supplied and the TagInput is rendered within the Popover,
+                        // without matchTargetWidth there is no width defined in any of TagInput's
+                        // grandparents when it's rendered through usePortal, so it will never flex-wrap
+                        // and infinitely grow horizontally. To address this, if there is no width guidance
+                        // from matchTargetWidth, explicitly set a default width to so Tags will flex-wrap.
```

Comment:
nice - I don't think we want to add this class when `matchTargetWidth` prop is set though

![Screenshot 2024-07-19 at 4 27 22 PM](https://github.com/user-attachments/assets/7ca0777a-4f15-4d52-8eb7-ceb56286e665)